Managers/UTMVirtualMachine+Drives.m

```diff
@@ -63,42 +70,62 @@ @implementation UTMVirtualMachine (Drives)
 
 - (BOOL)ejectDrive:(UTMDrive *)drive force:(BOOL)force error:(NSError * _Nullable __autoreleasing *)error {
     [self.viewState removeBookmarkForRemovableDrive:drive.name];
-    if (!self.qemu.isConnected) {
-        return YES; // not ready yet
+    [self.configuration setImagePath:@"" forIndex:drive.index];
+    bool saved = [self saveUTMWithError:error];
+    if (!saved) {
+        return NO;
+    }
+    if (self.qemu.isConnected) {
+        return [self.qemu ejectDrive:drive.name force:force error:error];
+    } else {
+        // not running
+        drive.status = UTMDriveStatusEjected;
+        return YES;
     }
-    return [self.qemu ejectDrive:drive.name force:force error:error];
 }
 
 - (BOOL)changeMediumForDrive:(UTMDrive *)drive url:(NSURL *)url error:(NSError * _Nullable __autoreleasing *)error {
-    NSData *bookmark = [url bookmarkDataWithOptions:0
+    // assume security scoped, it doesn't hurt even if the file is in the documents directory!
+    BOOL securityScoped = YES;
+    if (securityScoped && ![url startAccessingSecurityScopedResource]) {
+        return NO;
+    }
+    NSUInteger createOptions = securityScoped ? ( 1 << 11 ) : 0;
```

Comment:
I don’t think this flag is supported on iOS, you need minimal bookmark. https://developer.apple.com/documentation/uikit/view_controllers/providing_access_to_directories

Reply:
I was wondering if it was supported, as it is only documented on macOS. Thanks for that link!